FrontEnd/src/Components/NavBar.jsx

```diff
@@ -1,7 +1,19 @@
 import { Box, Flex, Button, Link } from "@chakra-ui/react";
+import { useAuth0 } from "@auth0/auth0-react";
+import { useEffect, useState } from "react";
 import logo from "./assets/logo1.svg";
 
 const NavBar = () => {
+  const [button1, setButtonValue] = useState("");
+  const { loginWithRedirect, isAuthenticated, logout } = useAuth0();
+  useEffect(() => {
+    if (isAuthenticated) {
+      setButtonValue("Logout");
+    } else {
+      setButtonValue("Register / Login");
+    }
+  }, [isAuthenticated]);
```

Comment:
The code here can be optimized by removing the use of additional state and have the conditional rendering based on `isAuthenticated` in the `render` method or `return` jsx itself

Reply:
Got it, will make changes 👍